CatBreeds/Model/CatImageHandler.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  CatImageHandler.swift
+//  CatBreeds
+//
+//  Created by Areeb Durrani on 4/10/25.
+//
+
+import Foundation
+
+struct Thumbnail: Decodable {
+    let source: String
+}
+
+struct Page: Decodable {
+    let thumbnail: Thumbnail
+}
+
+struct Query: Decodable {
+    let pages: [Page]
+}
+
+struct ImageResponse: Decodable {
+    let query: Query
+}
+
+func removePunctuation(from string: String, delimiter: Character) -> String {
+    var s = string
+    if s.contains(delimiter) {
+        s.removeSubrange(s.firstIndex(of: delimiter)!..<s.endIndex)
+    }
+    return s
+}
+
+func sanitizeInput(breed: String, addSuffix: Bool) -> String {
+    var toReturn = breed.replacingOccurrences(of: " ", with: "_")
+    if toReturn.contains("Persian") && toReturn.contains("Traditional") {
+        toReturn = "Traditional_Persian"
+    }
+    toReturn = removePunctuation(from: toReturn, delimiter: "(")
+    toReturn = removePunctuation(from: toReturn, delimiter: ",")
+    toReturn = removePunctuation(from: toReturn, delimiter: "[")
+    if toReturn.contains("Cymric") {
+        toReturn = "Cymric"
+    }
+    if toReturn.contains("Cheetoh") {
+        toReturn = "Bengal"
+    }
+    if toReturn.contains("Sam_Sawet") {
+        toReturn = "Thai"
+    }
+    if addSuffix {
+        if !toReturn.lowercased().hasSuffix("_cat") {
+            toReturn += "_cat"
+        }
+    }
+    return toReturn
+}
+
+func fetchImageResponse(breedName: String) async -> ImageResponse {
+    let url = URL(
+        string:
+            "https://en.wikipedia.org/w/api.php?action=query&format=json&formatversion=2&titles=\(breedName)&prop=pageimages|pageterms&pithumbsize=300&redirects=1"
+    )!
+    var received: ImageResponse = ImageResponse(query: Query(pages: []))
```

Comment:
Now you have a second url. Define all the urls in one place and use it. This will be useful in future to update urls when the end point changes